knora/excel2xml.py

```diff
@@ -0,0 +1,2127 @@
+    ALLOW = E.allow
+    # lxml.builder.E is a more sophisticated element factory than etree.Element.
+    # E.tag is equivalent to E("tag") and results in <tag>
+
+    res_default = etree.Element("{%s}permissions" % (xml_namespace_map[None]), id="res-default")
+    res_default.append(ALLOW("V", group="UnknownUser"))
+    res_default.append(ALLOW("V", group="KnownUser"))
+    res_default.append(ALLOW("CR", group="Creator"))
+    res_default.append(ALLOW("CR", group="ProjectAdmin"))
+    root_element.append(res_default)
+
+    res_restricted = PERMISSIONS(id="res-restricted")
+    res_restricted.append(ALLOW("RV", group="UnknownUser"))
+    res_restricted.append(ALLOW("V", group="KnownUser"))
+    res_restricted.append(ALLOW("CR", group="Creator"))
+    res_restricted.append(ALLOW("CR", group="ProjectAdmin"))
+    root_element.append(res_restricted)
+
+    prop_default = PERMISSIONS(id="prop-default")
+    prop_default.append(ALLOW("V", group="UnknownUser"))
+    prop_default.append(ALLOW("V", group="KnownUser"))
+    prop_default.append(ALLOW("CR", group="Creator"))
+    prop_default.append(ALLOW("CR", group="ProjectAdmin"))
+    root_element.append(prop_default)
+
+    prop_restricted = PERMISSIONS(id="prop-restricted")
+    prop_restricted.append(ALLOW("RV", group="UnknownUser"))
+    prop_restricted.append(ALLOW("V", group="KnownUser"))
+    prop_restricted.append(ALLOW("CR", group="Creator"))
+    prop_restricted.append(ALLOW("CR", group="ProjectAdmin"))
+    root_element.append(prop_restricted)
+
+    return root_element
+
+
+def make_resource(
+    label: str,
+    restype: str,
+    id: str,
+    permissions: str = "res-default",
+    ark: Optional[str] = None,
+    iri: Optional[str] = None
+) -> etree._Element:
+    """
+    Creates an empty resource element, with the attributes as specified by the arguments
+
+    Args:
+        The arguments correspond 1:1 to the attributes of the <resource> element.
+
+    Returns:
+        The resource element, without any children, but with the attributes
+        ``<resource label=label restype=restype id=id permissions=permissions ark=ark iri=iri></resource>``
+
+    Examples:
+        >>> resource = make_resource(...)
+        >>> resource.append(make_text_prop(...))
+        >>> root.append(resource)
+
+    See https://docs.dasch.swiss/latest/DSP-TOOLS/dsp-tools-xmlupload/#describing-resources-with-the-resource-element
+    """
+
+    kwargs = {
+        "label": label,
+        "restype": restype,
+        "id": id,
+        "permissions": permissions,
+        "nsmap": xml_namespace_map
+    }
+    if ark:
+        kwargs["ark"] = ark
+    if iri:
+        kwargs["iri"] = iri
+    if ark and iri:
+        handle_warnings(f"Both ARK and IRI were provided for resource '{label}' ({id}). The ARK will override the IRI.")
+
+    resource_ = etree.Element(
+        "{%s}resource" % (xml_namespace_map[None]),
+        **kwargs
+    )
+    return resource_
+
+
+def make_bitstream_prop(
```

Comment:
could all the `make_XX_prop()` functions be moved to a separate file? we're already at line 600 here...